falcon/response.py

```diff
@@ -202,6 +204,10 @@ def data(self):
     def data(self, value):
         self._data = value
 
+    @property
+    def headers(self):
+        return [header for header, value in self._headers.items()]
```

Comment:
Do we really need to iterate *items* just to create a list?
Also note that `items()` does a full copy on PY2, so we would be allocation twice completely needlessly.

Could we potentially simplify as:

```python
    @property
    def headers(self):
        return list(self._headers)
```

(Or *tuple(...)*, I have not give much thought about that).

Reply:
Didn't realize items did a full copy. Will change. Great catch!

Reply:
I have briefly gone through with what we have in other Falcon *Request* / *Response* properties and ``list`` seems more in line with the existing stuff, so lets go with it.

``tuple(dict argument)`` is, on the other hand, about 15-20% faster. But considering this is a property for information purposes / user convenience rather than a crucial piece used in forming every response, performance is not that important to optimize in this case. Unless @kgriffs has any insights on this.